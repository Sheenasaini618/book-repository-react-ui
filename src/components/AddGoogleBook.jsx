import * as React from 'react';
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogTitle from '@mui/material/DialogTitle';
import useMediaQuery from '@mui/material/useMediaQuery';
import { useTheme } from '@mui/material/styles';
import Box from '@mui/material/Box';
import Fab from '@mui/material/Fab';
import AddIcon from '@mui/icons-material/Add';
import { addGoogleBook } from '../util/api/bookApi';

export default function AddGoogleBook(props) {

  var bookData = props.details

  const [open, setOpen] = React.useState(false);
  const theme = useTheme();
  // const fullScreen = useMediaQuery(theme.breakpoints.down('md'));

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleGoogleBookAdd = () => {
    addGoogleBook(bookData).then((res) => {
      console.log(res);
      handleClose();
      window.history.go("http://localhost:3000/add");
    });
  };

  return (
    <div>
      <Box sx={{ '& > :not(style)': { m: 1 } }}>
        <Fab size="small" color="primary" aria-label="add" onClick={handleClickOpen}>
          <AddIcon />
        </Fab>
        <Dialog
          open={open}
          onClose={handleClose}
          aria-labelledby="responsive-dialog-title"
        >
          <DialogTitle id="responsive-dialog-title">
            {"Are you sure you want to add the book?"}
          </DialogTitle>
          <DialogActions>
            <Button onClick={handleClose}>
              No
            </Button>
            <Button onClick={handleGoogleBookAdd} autoFocus>
              Yes
            </Button>
          </DialogActions>
        </Dialog>
      </Box>
    </div>
  );
}