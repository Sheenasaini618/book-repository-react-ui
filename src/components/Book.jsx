import * as React from 'react';
import Card from '@mui/material/Card';
import CardHeader from '@mui/material/CardHeader';
import CardMedia from '@mui/material/CardMedia';
import CardContent from '@mui/material/CardContent';
import CardActions from '@mui/material/CardActions';
import Avatar from '@mui/material/Avatar';
import Typography from '@mui/material/Typography';
import { red } from '@mui/material/colors';
import ManualEditBook from './ManualEditBook';
import DeleteBook from './DeleteBook';


export default function Book(props) {

  var book = props.details
  console.log(book)
  return (
    <Card sx={{ maxWidth: 450, maxHeight: 500 }} style={{ marginTop: "10px" }}>
      <CardHeader
        avatar={
          <Avatar sx={{ bgcolor: red[500] }} aria-label="recipe">
            {book.title.charAt(0).toUpperCase()}
          </Avatar>
        }
        title={book.title}
        subheader={book.authors[0]}
      />
      <CardMedia
        component="img"
        sx={{ width: 151 }}
        height="194"
        image={book.imageLinks.smallThumbnail}
        alt="Paella dish"
      />
      <CardContent>
        <Typography variant="h6" color="text.secondary" align='justify' style={{ fontWeight: 600 }}>
          Price: â‚¹{book.price}.00
        </Typography>
        <Typography variant="h6" color="text.secondary" align='justify' style={{ fontWeight: 600 }}>
          Copies Available: {book.quantity}
        </Typography>
      </CardContent>
      <CardActions disableSpacing>
        <ManualEditBook details={book} />
        <DeleteBook details={book.id} />
      </CardActions>
    </Card>
  );
}
