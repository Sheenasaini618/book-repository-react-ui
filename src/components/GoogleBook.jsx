import * as React from 'react';
import Card from '@mui/material/Card';
import CardHeader from '@mui/material/CardHeader';
import CardMedia from '@mui/material/CardMedia';
import CardContent from '@mui/material/CardContent';
import CardActions from '@mui/material/CardActions';
import Avatar from '@mui/material/Avatar';
import Typography from '@mui/material/Typography';
import { red } from '@mui/material/colors';
import AddGoogleBook from './AddGoogleBook';


export default function GoogleBook(props) {

  var book = props.details
  
  console.log(book)
  return (
    <Card >
      <CardHeader 
        avatar={
          <Avatar sx={{ bgcolor: red[500] }} aria-label="recipe">
            {book.volumeInfo.title.charAt(0).toUpperCase()}
          </Avatar>
        }
        title= {book.volumeInfo.title}
        subheader= {book.volumeInfo.authors}
      />
       <CardMedia
        component="img"
        sx={{ width: 151 }}
        height="194"
        image= {book.volumeInfo.imageLinks.smallThumbnail}
        alt="Paella dish"
      /> 
      <CardContent>
        {/* <Typography variant="body2" color="text.secondary" align='justify' style={{height : 70}}>
        {book.volumeInfo.description}
        </Typography> */}
        <Typography variant="h6" color="text.secondary" align='justify' style={{fontWeight : 600}}>
        </Typography>
        <Typography variant="h6" color="text.secondary" align='justify' style={{fontWeight : 600}}>
        Copies Available: {book.volumeInfo.quantity}
        </Typography>
      </CardContent>
      <CardActions disableSpacing>
      <AddGoogleBook details={book} />
      </CardActions>
    </Card>
  );
}
