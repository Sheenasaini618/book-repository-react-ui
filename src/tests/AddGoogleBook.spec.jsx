import React from 'react'
import Book from '../components/Book'
import '@testing-library/jest-dom'
import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow, render } from 'enzyme'
import { configure } from 'enzyme';
import GoogleBook from '../components/GoogleBook';
import AddGoogleBook from '../components/AddGoogleBook';


configure({ adapter: new Adapter() });
describe('<AddGoogleBook />', () => {

    let props, component, wrapper;

    beforeEach(() => {
        // charAt: jest.fn()

        props = {
            details: {
                "volumeInfo": {
                  "id": 1,
                  "title": "Introduction to Probability",
                  "imageLinks": {
                    "smallThumbnail": "http://books.google.com/books/content?id=2HxYV9lWlEcC&printsec=frontcover&img=1&zoom=5&source=gbs_api",
                    "thumbnail": "http://books.google.com/books/content?id=2HxYV9lWlEcC&printsec=frontcover&img=1&zoom=1&source=gbs_api"
                  },
                  "authors": [
                    "John E. Freund"
                  ],
                  "description": "Featured topics include permutations and factorials, probabilities and odds, frequency interpretation, mathematical expectation, decision making, postulates of probability, rule of elimination, much more. Exercises with some solutions. Summary. 1973 edition.",
                  "price": 12566,
                  "quantity": 3
                }
              }
        }

        component = shallow(<AddGoogleBook {...props} />);
    })

    it('should render book ', () => {
        const element = component.debug()
        expect(element).toMatchSnapshot();
    });
})