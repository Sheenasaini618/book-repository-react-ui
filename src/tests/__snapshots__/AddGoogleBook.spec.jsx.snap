// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddGoogleBook /> should render book  1`] = `
"<div>
  <ForwardRef(Box) sx={{...}}>
    <ForwardRef(Fab) size=\\"small\\" color=\\"primary\\" aria-label=\\"add\\" onClick={[Function: handleClickOpen]}>
      <Memo(ForwardRef(AddIcon)) />
    </ForwardRef(Fab)>
    <ForwardRef(Dialog) open={false} onClose={[Function: handleClose]} aria-labelledby=\\"responsive-dialog-title\\">
      <ForwardRef(DialogTitle) id=\\"responsive-dialog-title\\">
        Are you sure you want to add the book?
      </ForwardRef(DialogTitle)>
      <ForwardRef(DialogActions)>
        <ForwardRef(Button) onClick={[Function: handleClose]}>
          No
        </ForwardRef(Button)>
        <ForwardRef(Button) onClick={[Function: handleGoogleBookAdd]} autoFocus={true}>
          Yes
        </ForwardRef(Button)>
      </ForwardRef(DialogActions)>
    </ForwardRef(Dialog)>
  </ForwardRef(Box)>
</div>"
`;
